cli/cmd/main.go

```diff
@@ -227,6 +236,123 @@ func (c *tagsCmd) Run() error {
+}
+
+func (c *simulateCmd) Run() error {
+	parser := parsers.NewEarthlyParser()
+	scanner := scanners.NewFileScanner([]string{c.Path}, parser, afero.NewOsFs())
+
+	fmt.Println("Targets: ", c.Targets)
+	// Loop through target patterns.
+	for _, tp := range strings.Fields(c.Targets[0]) {
+		fmt.Println(">>>>>>> Detecting", tp, "target")
+		pathToEarthMap, err := scanner.ScanForTarget(tp)
+		if err != nil {
+			return err
+		}
+		for _, e := range pathToEarthMap {
+			var wg sync.WaitGroup
+			// Loop through filtered targets.
+			for _, tg := range e.Targets {
+				target := filepath.Join(filepath.Dir(e.Earthfile.Path), "+"+tg)
+				fmt.Println(">>> Running target", target)
+				wg.Add(1)
+				go runEarthlyTarget(&wg, target)
+			}
+			// Wait for all goroutines to finish.
+			wg.Wait()
+		}
+
+		fmt.Println("Command executed successfully.")
+	}
+	return nil
+}
+
+// Run Earthly with target.
+func runEarthlyTarget(wg *sync.WaitGroup, earthlyCmd string) {
+	defer wg.Done()
+	command := "earthly"
+
+	// Create the command.
+	cmd := exec.Command(command, earthlyCmd)
+
+	var stdoutBuf, stderrBuf bytes.Buffer
+	cmd.Stdout = io.MultiWriter(os.Stdout, &stdoutBuf)
+	cmd.Stderr = io.MultiWriter(os.Stderr, &stderrBuf)
+
+	err := cmd.Run()
+	if err != nil {
+		log.Fatalf("cmd.Run() failed with %s\n", err)
+	}
+	outStr, errStr := stdoutBuf.String(), stderrBuf.String()
+	fmt.Printf("\nout:\n%s\nerr:\n%s\n", outStr, errStr)
+}
+
+type generateCmd struct {
+	Path      string   `                      help:"directory path to be iterated to search for targets within the Earthfile"               arg:"" type:"path"`
+	Targets   []string `short:"t"             help:"Earthly targets pattern"                                               default:"check,check-*,build,test,test-*"`
+	Version   string   `short:"v"             help:"Earthly version"                    default:"0.7"`
+	Directory string   `short:"d"             help:"Directory to create a Earthfile"    default:"generate"`
+}
+
+// Generate Earthfile with given targets.
+// All targets associated with the given targets will be listed inside.
+func (c *generateCmd) Run() error {
+	// Directory path and file name.
+	fileName := "Earthfile"
+
+	// Create the directory.
+	err := os.MkdirAll(c.Directory, os.ModePerm)
+	if err != nil {
+		return nil
+	}
+
+	filePath := filepath.Join(c.Directory, fileName)
+	file, err := os.Create(filePath)
+	if err != nil {
+		return nil
+	}
+
+	writer := bufio.NewWriter(file)
+	// Write down Earthly version and main target.
+	setup := fmt.Sprintf("VERSION --global-cache %s \nsimulate:\n", c.Version)
+	_, err = writer.WriteString(setup)
+	if err != nil {
+		return nil
+	}
+
+	parser := parsers.NewEarthlyParser()
+	scanner := scanners.NewFileScanner([]string{c.Path}, parser, afero.NewOsFs())
+	// Loop through target patterns.
```

Comment:
Same as above.  We really should have a function which gets the sorted list of targets, and then we should iterate that to create the earthfile.  In this case, and the case above the function to get the target list should be the same.